backend/src/main/java/com/github/astridottmann/backend/services/CompareRoutesService.java

```diff
@@ -35,11 +36,9 @@ public CompareRoutes getCompareRoutesById(String id) {
                 .orElseThrow(() -> new NoSuchElementException("Not found!"));
     }
 
-    public CompareRoutes addComparison(List<Route> compared) {
-        CompareRoutes compareRoutesToAdd = new CompareRoutes(
-                idService.createRandomId(),
-                List.of(compared.get(0), compared.get(1)),
-                compareEmissions(compared));
+    public CompareRoutes addComparison(CompareRoutesDTO compareRoutesDTO) {
+        CompareRoutes compareRoutesToAdd =
+                CompareRoutes.createCompareRoutesFromDTO(compareRoutesDTO, idService.createRandomId(), compareEmissions(compareRoutesDTO.compared()));
```

Comment:
optional: remove the idService an put it inside the method. so you can't forget to create a random id.
hint: I think this will be a bigger impact because you have to rewrite your tests

Reply:
thanks for the hint ;-)